src/shogun/base/class_list.h

```diff
@@ -37,19 +39,28 @@ namespace shogun {
 	 *
 	 */
 	template <class T>
-	T* create_object(const char* name)
+	T* create_object(
+	    const char* name,
+	    EPrimitiveType pt = PT_NOT_GENERIC) throw(ShogunException)
 	{
-		auto* object = create(name, PT_NOT_GENERIC);
+		auto* object = create(name, pt);
 		if (!object)
 		{
-			SG_SERROR("No such class %s", name);
+			SG_SERROR(
+			    "Class %s with primitive type %s does not exist.\n", name,
```

Comment:
IMO we need std::iostream for our logs, then we could just overload the << operator and dont have to introduce all these to_string methods ...

Reply:
although it looks shitty but `operator <<` will not solve your problem to convert some objects to string.. that only works with primitive types

Reply:
on the other hand having these macros is nice in a long run as we could switch finally our logging backend. maybe spdlog? that allows us to redirect the logs into any `sink` not only stdout/err

Reply:
I know but I can overload the << thing, no? Then I dont need the function calls when I write error messages, but can collect them somewhere else?
Totally up for doing a new logging backend. GSoC project part?

Reply:
new logging backend = just change the macro definitions.
yeah you can offload the operator << but that means that you force to use `std::iostream`-like interface to be used, which is not the most liked IO back-end (for good reasons) in c++ community

Reply:
on the other hand you can do the same thing, i.e. implement to_string() for that object where you wanna have operator <<, right?

Reply:
I think we can introduce an std::cout-like object which supports <<. Internally it could use any logger we want.